lib/services/wallet_connect_service.dart

```diff
@@ -334,19 +342,45 @@ class WalletConnectService {
             event =
                 RequestSendQubicEvent.fromMap(args, topic, sessionRequest.id);
             event.validateOrThrow();
-
-            if (web3Wallet!.getActiveSessions().containsKey(topic)) {
-              event.setPairingMetadata(
-                  web3Wallet!.getActiveSessions()[topic]!.peer.metadata);
-            } else {
-              throw "Session not found";
+            validateAndSetSession(topic, event);
+            return web3Wallet!.respondSessionRequest(
+                topic: topic,
+                response: JsonRpcResponse(
+                    id: sessionRequest.id,
+                    result: await sendQubicHandler!(event)));
+          } catch (e) {
+            if (e is JsonRpcError) {
+              rethrow;
             }
+            throw JsonRpcError(code: -2, message: e.toString());
+          }
+        });
 
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wSendTransaction,
+        handler: (topic, args) async {
+          final sessionRequest = web3Wallet!.pendingRequests
+              .getAll()
+              .where((e) =>
+                  e.method == WcMethods.wSendTransaction && e.topic == topic)
+              .last;
```

Comment:
this lines above is also common for all the request handles and can be move to a method

Reply:
@sallymoc 

Fixed with 0845880